Meadowlark-js/backends/meadowlark-aws-dynamodb-opensearch/src/BaseDynamoRepository.ts

```diff
@@ -48,6 +49,23 @@ export function getDynamoDBDocumentClient(): DynamoDBDocumentClient {
   return dynamoDBDocumentClient;
 }
 
+export function buildNKString(naturalKey: string): string {
+  if (naturalKey.startsWith('NK#')) return naturalKey;
+
+  return `NK#${naturalKey}`;
+}
+
+/**
+ * Converts document identity to string in DynamoDB natual key form
+ * For example, converts:
+ * [{name: 'classPeriodName', value: 'z1'}, {name: 'schoolId', value: '24'}, {name: 'studentId', value: '333'}]
+ * to 'NK#classPeriodName=z1#schoolId=24#studentId=333'
+ */
+export function dynamoIdentityToString(documentIdentity: DocumentIdentity): string {
+  const stringifiedValues: string[] = documentIdentity.map((element: DocumentElement) => `${element.name}=${element.value}`);
```

Comment:
deferring NK/FK strings creation, moving to dynamo backend

Reply:
Correct. Now that document identities are hashed to the document id, all we need across different data stores are the document ids.